STIPS/STIP-009.md

```diff
@@ -0,0 +1,743 @@
+
+**Reviewer**:
+
+Reviewer: []
+
+## Specification
+
+### ManagerFactory
+
+#### Events
+
+##### DelegatedManagerDeployed
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|setToken|Address of SetToken to be managed|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the DelegatedManager|
+
+
+##### DelegatedManagerInitialized
+
+| Type  | Name  | Description   |
+|------ |------ |-------------  |
+|address|setToken|Address of SetToken to be managed|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the DelegatedManager|
+
+
+#### Structs
+
+##### InitializeParams
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|deployer|Address of the deployer|
+|address|owner|Address of the owner|
+|address|manager|Address of the DelegatedManager|
+|bool|isPending|Bool if manager in pending state|
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|factory|Address of Set Token factory|
+|mapping(address => InitializeParams)|initialize|Mapping from Set Token to initialization parameters|
+
+
+#### Functions
+
+| Name  | Caller  | Description 	|
+|------	|------	|-------------	|
+|createSetAndManager |deployer|Create new Set Token with a DelegatedManager manager|
+|createManager | SetToken owner |Migrate existing Set Token to a DelegatedManager manager|
+|initialize |deployer or SetToken owner |Initialize modules and extensions|
+
+> createSetAndManager
+
+```solidity
+function createSetAndManager(
+    address[] memory _components,
+    uint256 _units,
+    string memory _name,
+    string memory _symbol,
+    address _owner,
+    address _methodologist,
+    address[] memory _modules,
+    address[] memory _operators,
+    address[] memory _assets,
+    address[] memory _extensions
+)
+    external
+{
+    if (_assets.length != 0) {
+        _validateComponentsMatchWhitelistedAssets(_components, assets);
+    }
+
+    address setTokenAddress = _deploySet(
+        _components,
+        _modules,
+        _units,
+        _name,
+        _symbol
+    );
+
+    address managerAddress = _deployManager(
+        setTokenAddress,
+        _methodologist,
+        _operators,
+        _assets,
+        _extensions
+    );
+
+    initialize[setTokenAddress] = InitializeParams({
+        deployer: msg.sender,
+        owner: _owner,
+        manager: managerAddress,
+        isPending: true
+    });
+}
+```
+
+> createManager
+```solidity
+function createManager(
+    address memory _setTokenAddress,
+    address _owner,
+    address _methodologist,
+    address[] memory _modules,
+    address[] memory _operators,
+    address[] memory _assets,
+    address[] memory _extensions
+)
+    external
+{
+    _validateComponentsMatchWhitelistedAssets(setToken.getComponents(), assets);
+
+    address managerAddress = _deployManager(
```

Comment:
Let's either also add the pseudo-code for this internal function or at least note that in here we have logic that decides whether to toggle the asset allow list on/off (empty array off/filled array on)